### ISSUE TITLE
Way to set a serviceaccount for the integration pod?

### ISSUE THREAD

**Issue by dylanenabled** [apache/camel-k#1030]
I'm using AWS EKS to host a kubernetes cluster and I want my camelk integration to connect to an AWS SQS queue to receive messages. 

AWS recently released [IAM roles for service accounts ](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) which lets me attach a role to serviceaccounts with policies that allow the role to access the SQS service. (It does this by injecting environment variables and a secret volume into the pod containers with that service account attached). Given a recent AWS SDK, this will then pick up the role from the environment to create tokens to access SQS securely.

So, basically, is there a way to tell `kamel run` to attach a specific serviceaccount to the integration pod it creates?

**Comment by himanshumps** [comment]
I am interested in knowing how to do this as well

**Comment by Richardmbs12** [comment]
This is also applicable to the camel k helm chart values file attributes.
The service account in the helm chart can be enhanced as follows:

Current:

```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: camel-k-operator
  labels:
    app: "camel-k"
    {{- include "camel-k.labels" . | nindent 4 }}
```
  
To Be:

```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: camel-k-operator
  labels:
    app: "camel-k"
    {{- include "camel-k.labels" . | nindent 4 }}
  annotations:
    {{- include "service-account.annotations" . | nindent 4 }}
```

This dynamic "annotations" config can be then beneficial for iam role annotations, or any other annotation valuableto be added to a service account, customizable via values file

**Comment by squakez** [comment]
Duplicated in #3797
